src/deploy/hosting/prepare.js

```diff
@@ -67,6 +69,21 @@ module.exports = function(context, options) {
       );
     }
 
+    if (cfg.i18n) {
+      if (!cfg.i18n.root) {
+        throw new FirebaseError("The root in the i18n config can't be empty.");
+      } else {
+        const i18nPath = path.join(cfg.public, cfg.i18n.root);
+        if (!fsutils.dirExistsSync(resolveProjectPath(options, i18nPath))) {
+          utils.logLabeledWarning(
+            "hosting",
+            `The I18n Rewrites feature is enabled, but the directory ` +
+              `could not be found within '${cfg.public}' at root '${cfg.i18n.root}'.`
```

Comment:
Note: applying this suggestion will also require importing bold from `cli-color`.

```suggestion
            `Couldn't find specified i18n root directory ${bold(cfg.i18n.root)} in public directory ${bold(cfg.public)}.`
```

Reply:
Done.